src/transformers/models/mamba2/modeling_mamba2.py

```diff
@@ -1373,7 +1373,7 @@ def slow_forward(self, hidden_states, cache):
             )
 
             # Subsequent output
-            y = torch.einsum("bhpn,bn->bhp", cache.ssm_states[self.layer_idx], C)
+            y = torch.einsum("bhpn,bn->bhp", cache.ssm_states[self.layer_idx].to(C.dtype), C)
```

Comment:
We save the cache in fp32 but we compute in the current type, meaning we need to reconvert on computation :D